src/algebraic_topology/alternating_face_map_complex.lean

```diff
@@ -0,0 +1,240 @@
+/-
+Copyright (c) 2021 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebra.homology.homological_complex
+import algebraic_topology.simplicial_object
+import algebraic_topology.Moore_complex
+import category_theory.abelian.basic
+import algebra.big_operators.basic
+import tactic.ring_exp
+import data.fintype.card
+
+/-!
+
+# The alternating face map complex of a simplicial object in a preadditive category
+
+We construct the alternating face map complex, as a
+functor `alternating_face_map_complex : simplicial_object C ⥤ chain_complex C ℕ`
+for any preadditive category `C`. For any simplicial object `X` in `C`,
+this is the homological complex `... → X_2 → X_1 → X_0`
+where the differentials are alternating sums of faces.
+
+We also construct the natural transformation
+`inclusion_of_Moore_complex : normalized_Moore_complex A ⟶ alternating_face_map_complex A`
+when `A` is an abelian category.
+
+## References
+* https://stacks.math.columbia.edu/tag/0194
+* https://ncatlab.org/nlab/show/Moore+complex
+
+-/
+
+open category_theory category_theory.limits category_theory.subobject
+open opposite
+
+open_locale big_operators
+open_locale simplicial
+
+noncomputable theory
+
+namespace algebraic_topology
+
+namespace alternating_face_map_complex
+
+/-!
+## Construction of the alternating face map complex
+-/
+
+/-- In degree n, the alternating face map complex is given by
+the nth-object of the simplicial object -/
+@[simp]
+def obj_X {C : Type*} [category C] (X : simplicial_object C) (n : ℕ) := X _[n]
```

Comment:
I don't think there is much value in this def'n now. In the end `X _[n]` is shorter than `obj_X X n`.

Reply:
Indeed, I have just removed it.